airflow/jobs.py

```diff
@@ -2193,19 +2235,124 @@ def _execute(self):
                     dep_context=DepContext(ignore_depends_on_past=True),
                     session=session,
                     verbose=True)
-                for t in deadlocked)
+                for t in ti_status.deadlocked)
             if deadlocked_depends_on_past:
                 err += (
                     'Some of the deadlocked tasks were unable to run because '
                     'of "depends_on_past" relationships. Try running the '
                     'backfill with the option '
                     '"ignore_first_depends_on_past=True" or passing "-I" at '
                     'the command line.')
-            err += ' These tasks have succeeded:\n{}\n'.format(succeeded)
-            err += ' These tasks have started:\n{}\n'.format(started)
-            err += ' These tasks have failed:\n{}\n'.format(failed)
-            err += ' These tasks are skipped:\n{}\n'.format(skipped)
-            err += ' These tasks are deadlocked:\n{}\n'.format(deadlocked)
+            err += ' These tasks have succeeded:\n{}\n'.format(ti_status.succeeded)
+            err += ' These tasks have started:\n{}\n'.format(ti_status.started)
+            err += ' These tasks have failed:\n{}\n'.format(ti_status.failed)
+            err += ' These tasks are skipped:\n{}\n'.format(ti_status.skipped)
+            err += ' These tasks are deadlocked:\n{}\n'.format(ti_status.deadlocked)
+
+        return err
+
+    def _get_run_dates(self):
+        # Get intervals between the start/end dates, which will turn into dag runs
+        run_dates = []
+
+        start_date = self.bf_start_date
+        end_date = self.bf_end_date
+
+        # create dag run ids / dates
+        dr_start_date = start_date or min([t.start_date for t in self.dag.tasks])
+        end_date = end_date or datetime.now()
+
+        # next run date for a subdag isn't relevant (schedule_interval for subdags
+        # is ignored) so we use the dag run's start date in the case of a subdag
+        next_run_date = (self.dag.normalize_schedule(dr_start_date)
+                         if not self.dag.is_subdag else dr_start_date)
+
+        while next_run_date and next_run_date <= end_date:
+            run_dates.append(next_run_date)
+            next_run_date = self.dag.following_schedule(next_run_date)
+
+        return run_dates
+
+    @provide_session
+    def _execute_for_run_dates(self, run_dates, ti_status, executor, pickle_id,
```

Comment:
Would be good to make it more clear what the difference between this method and _execute is (either better method names or comments).